### PR TITLE
refactor logging package and add helpers

### PR DESCRIPTION
**Pull request by vxio**
Added methods:
- `logger.Log(msg string)` which calls `logger.Logf(format string, args ...string)`
- `logger.LogError(err error) error` which calls `logger.LogErrorf(format string, args ...string)`

And moved interfaces and helper items to `logger.go` file to simplify package layout.

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -9,6 +15,59 @@ type Logger interface {
 	Error() Logger
 	Fatal() Logger
 
-	Logf(format string, a ...interface{})
-	LogErrorf(format string, a ...interface{}) error
+	Log(message string)
+	Logf(format string, args ...interface{})
```
Comment (vxio) [inline log/logger.go:19]: We could use `Logf(format string, args ...interface{})` for cases where we'd want to pass in a string without args, but adding `Log(msg string)` avoids any subtle bugs where the caller meant to pass in an interpolated string but forgot to add in the args.

File: log/logger_test.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func Test_Caller(t *testing.T) {
 	a, buffer, log := Setup(t)
 
 	log.Info().With(StackTrace).Logf("message")
-	a.Regexp(regexp.MustCompile(`caller_0=(.*?)(\/log\/logger_test\.go)`), buffer.String())
+	a.Regexp(regexp.MustCompile(`caller_\d=(.*?)(\/log\/logger_impl\.go)`), buffer.String())
```
Comment (InfernoJJ) [inline log/logger_test.go:127]: We don't want to announce the logger_impl.go file. We want to filter this out.

Comment (InfernoJJ) [reply log/logger_test.go:127]: This is where the filter is: https://github.com/moov-io/base/pull/102/files#diff-330686dc447b6523e8cbb02ae01e63ed53f3b214274ecc0da55c978965dda5ddR63

**Review by InfernoJJ — CHANGES_REQUESTED** [review]

**Review by darwinz — APPROVED** [review]

**Review by alovak — APPROVED** [review]

File: log/logger_test.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func Test_Caller(t *testing.T) {
 	a, buffer, log := Setup(t)
 
 	log.Info().With(StackTrace).Logf("message")
-	a.Regexp(regexp.MustCompile(`caller_0=(.*?)(\/log\/logger_test\.go)`), buffer.String())
+	a.Regexp(regexp.MustCompile(`caller_\d=(.*?)(\/log\/logger_test\.go)`), buffer.String())
```
Comment (InfernoJJ) [inline log/logger_test.go:127]: We also want to make sure its `called_0` so the first element of the stack where log was called.

**Review by InfernoJJ — CHANGES_REQUESTED** [review]

**Review by InfernoJJ — APPROVED** [review]

**Pull request merged**